src/viking.js

```diff
@@ -1,11 +1,118 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(health, strength)
+        this.name = name
+    }
+
+    attack() {
+        super.attack()
+        return this.strength
+    }
+
+    receiveDamage(damage){
+        super.receiveDamage(damage)
+
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else {
+            return `${this.name} has died in act of combat`
+        }   
+    }
+
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength)
+    }
+    receiveDamage(damage) {
+        super.receiveDamage(damage)
+
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else {
+            return `A Saxon has died in combat`
+        }   
+    }
+}
 
 // War
-class War {}
+class War {
+    vikingArmy = []
+    saxonArmy = []
+
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon)
+    }
+
+    vikingAttack() {
+        const saxonI = Math.floor(Math.random() * this.saxonArmy.length)
+        const vikingI = Math.floor(Math.random() * this.vikingArmy.length)
+
+        const saxon = this.saxonArmy[saxonI]
+        const viking = this.vikingArmy[vikingI]
```

Comment:
Esto esta perfecto! (los nombres de las variables a lo mejor no me gustan mucho jejej pero bueno) ahora te digo se puede poner asi sacandolo todo a una constante o solo tener el resultado final! Me refiero:

```suggestion
        const saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)]
        const viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)]
```

Asi te lo encontrarias en el curro, siempre y cuando la linea no sea muy larga lo encontraras asi!